### ISSUE TITLE
`REMOTE_BUFFER_SIZE = 0` ==> Metrics discovery broken

### ISSUE THREAD

**Issue by Quentin-M** [graphite-project/graphite-web#2352]
Hi,

We recently upgraded our cluster to 1.1.4 in order to profit from https://github.com/graphite-project/graphite-web/pull/2314 and thus work around the potential 500x slowdown (was it not tested?) described in https://github.com/graphite-project/graphite-web/issues/2313#issuecomment-401876110. We then set `REMOTE_BUFFER_SIZE = 0` as documented.

But now, our clusters (populated or empty) fail to do metrics discovery for the most part. Here is an example:

```$ curl 'localhost:8080/metrics/find?from=1536617499&query=*&until=1536618399'

<body style="background-color: #666666; color: black;">
<center>
<h2 style='font-family: "Arial", sans-serif'>
<p>Graphite encountered an unexpected error while handling your request.</p>
<p>Please contact your site administrator if the problem persists.</p>
</h2>
<br/>
<div style="width: 50%; text-align: center; font-family: monospace; background-color: black; font-weight: bold; color: #ff4422;">

</div>

<div style="width: 70%; text-align: left; background-color: black; color: #44ff22; border: thin solid gray;">
<pre>
Traceback (most recent call last):
  File &quot;/opt/graphite/webapp/graphite/finders/remote.py&quot;, line 112, in find_nodes
    result, buffer_size=settings.REMOTE_BUFFER_SIZE), encoding=&#39;utf-8&#39;)
  File &quot;/opt/graphite/webapp/graphite/umsgpack.py&quot;, line 860, in _unpack3
    return _unpack(fp, options)
  File &quot;/opt/graphite/webapp/graphite/umsgpack.py&quot;, line 773, in _unpack
    code = _read_except(fp, 1)
  File &quot;/opt/graphite/webapp/graphite/umsgpack.py&quot;, line 589, in _read_except
    raise InsufficientDataException()
graphite.umsgpack.InsufficientDataException

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py&quot;, line 35, in inner
    response = get_response(request)
  File &quot;/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py&quot;, line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File &quot;/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py&quot;, line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File &quot;/opt/graphite/webapp/graphite/metrics/views.py&quot;, line 128, in find_view
    leaves_only=leaves_only,
  File &quot;/opt/graphite/webapp/graphite/storage.py&quot;, line 264, in find
    for match in self._find(query):
  File &quot;/opt/graphite/webapp/graphite/storage.py&quot;, line 295, in _find
    for node in result or []:
  File &quot;/opt/graphite/webapp/graphite/finders/remote.py&quot;, line 121, in find_nodes
    raise Exception(&quot;Error decoding find response from %s: %s&quot; % (result.url_full, err))
Exception: Error decoding find response from http://graphite-cache-1.graphite-cache.dev.svc.cluster.local:8080/metrics/find/?local=1&amp;format=msgpack&amp;query=%2A&amp;from=1536617499&amp;until=1536618399:

</pre>
</div>

</center>
```

```
$ curl 'http://graphite-cache-1.graphite-cache.dev.svc.cluster.local:8080/metrics/find/?local=1&amp;format=msgpack&amp;query=%2A&amp;from=1536617499&amp;until=1536618399'
���path�carbon�is_leaf�path�stats�is_leaf�~
```

Best Regards.

**Comment by deniszh** [comment]
Huh, I didn't realize that REMOTE_BUFFER_SIZE is used in find code too - 
https://github.com/graphite-project/graphite-web/blob/17d4804c961741d0562cb8d574ce9ed796240fb1/webapp/graphite/finders/remote.py#L110-L115
Not sure if it will work with reverted logic (when REMOTE_BUFFER_SIZE == 0 disables buffering) by default, am I right, @clusterfudge ?

BTW, according to my tests installing `python-msgpack` module making msgpack unpacking blazingly fast, even with default streaming deserialize.

**Comment by deniszh** [comment]
Fixed in master, will  be released in 1.1.5
